Lisää kävelyn ja pyöräilyn väylien ajoratakm laskenta

jarnovayrynen opened the pull request.

jarnovayrynen committed:
Lisää kävelyn ja pyöräilyn väylien ajoratakm laskenta

solita-staskila commented on src/clj/harja/kyselyt/tieverkko.sql:
```diff
@@ -224,7 +224,9 @@ ORDER BY tie, osa, alkuetaisyys;
                -- huomioitava vain pääkaistat 11 ja 21, muuten esim. kääntymiskaistoista tulee häiriötä laskentaan
                  (("tr-ajorata" = 0 AND "tr-kaista" = '11') OR
                   ("tr-ajorata" = 1 AND "tr-kaista" = '11') OR
-                  ("tr-ajorata" = 2 AND "tr-kaista" = '21'))
+                  ("tr-ajorata" = 2 AND "tr-kaista" = '21') OR
+                     -- joitakin kävelyn ja pyöräilyn väyliä on rajoitusalueina, niillä on aina tienumero 70000-80000, ajorata 0 ja kaista 31
+                  ("tr-numero" BETWEEN 70000 AND 80000 AND "tr-ajorata" = 0 AND "tr-kaista" = '31'))
```
Tässä ollaan postgresin varassa, että kanta castaa stringin integer muotoon koska columnit taulussa ovat int4 tyyppiä, joka näyttää tässä toimivan, mutta muuttaisin stringit '11' vaan integeriksi -> 11

jarnovayrynen replied:
No mutta hyvänen aika - näinhän se on. Kiitos, muuttakaamme integeriksi.

jarnovayrynen committed:
Jos ajoratakm laskenta epäonnistuu, käytä pituutta

On joitakin harvinaisia kohtia kaista-aineistossa (rampit), joista ei voida varmuudella päätellä pääkaistaa ja ajoratakilometrejä, kuten muualla. Näissä tapauksissa jää käytännössä ainoa ajorata nousematta SQL:stä. Tällöin laskennan tarkkuutta parantaa pituustiedon käyttö, joka antaa suurimmassa osassa tilanteita tai aina oikean vastauksen.

jarnovayrynen committed:
Käytä kaistojen vertailussa INTEGER-tyyppiä
